bdsf/islands.py

```diff
@@ -100,6 +105,26 @@ def __call__(self, img):
             img.nisl = len(img.islands)
             img.pyrank = det_img.pyrank
             img.minpix_isl = det_img.minpix_isl
+
+            if opts.output_all:
+                write_islands(img)
+            if opts.savefits_rankim or opts.output_all:
+                func.write_image_to_file(img.use_io, img.imagename + '_pyrank.fits', img.pyrank, img)
+            if opts.savefits_det_rmsim or opts.output_all:
+                resdir = img.basedir + '/background/'
+                if not os.path.exists(resdir):
+                    os.makedirs(resdir)
```

Comment:
Not sure if this execution path can be taken multiple times, but `os.makedirs` will raise an exception if the directory already exists. To avoid this, you need to add the option `exist_ok=True`.